src/Common/Core/Impl/Extensions/CultureExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Common.Core {
+    public static class CultureExtensions {
+        private static readonly char[] _separators = new char[] { ',', ' ' };
+
+        public static string LanguageNameFromLCID(int lcid) {
+            var culture = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures).FirstOrDefault(x => x.LCID == lcid);
+            return culture.GetLanguageName();
+        }
+
+        public static IEnumerable<string> GetLanguageNames() {
+            IEnumerable<CultureInfo> cultures = CultureInfo.GetCultures(CultureTypes.InstalledWin32Cultures);
+            return cultures.Select(c => c.GetLanguageName()).Distinct().OrderBy(x => x);
```

Comment:
Perhaps it is better to not deal with culture names at all here, and just show encoding names? Those are guaranteed to be unique; and given that we only set LC_CTYPE (and not all locale categories), showing culture name is somewhat misleading anyway...

And it appears that R's (actually msvcrt.dll) `setlocale` lets you specify only the encoding name like this: `Sys.setlocale(LC_CTYPE, ".1251")`

(see here for syntax: https://msdn.microsoft.com/en-us/library/aa246449(v=vs.60).aspx)